server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/Resend.java

```diff
@@ -323,4 +324,20 @@ protected String getMessage(Mail originalMail) throws MessagingException {
       return addresses.toArray(new InternetAddress[addresses.size()]);
     }
 
+    @Override
+    protected MailAddress getReplyTo() throws MessagingException {
+        String replyTo = getInitParameters().getReplyTo();
+        if (Strings.isNullOrEmpty(replyTo)) {
+            return null;
+        }
+
+        List<MailAddress> extractAddresses = AddressExtractor.withContext(getMailetContext())
+                .allowedSpecials(ImmutableList.of("postmaster", "sender", "null", "unaltered"))
+                .extract(replyTo);
+        if (extractAddresses.isEmpty()) {
```

Comment:
This pattern of returning first element seems recurent. Maybe you can add a **extractSingleAddress** method to AddressExtractor